atomic_reactor/plugins/build_orchestrate_build.py

```diff
@@ -389,26 +406,45 @@ def do_worker_build(self, release, cluster_info, koji_upload_dir, task_id):
                 except OsbsException:
                     pass
 
+    def select_and_start_cluster(self, platform):
+        ''' Choose a cluster and start a build on it '''
+
+        config = get_config(self.workflow)
+        clusters = config.get_enabled_clusters_for_platform(platform)
+
+        if not clusters:
+            raise UnknownPlatformException('No clusters found for platform {}!'
+                                           .format(platform))
+
+        cluster_fails = Counter()
+        retry_at = {}
+        possible_clusters = None
+
+        while True:
+            if not clusters:
+                raise RuntimeError('Could not find appropriate cluster for worker build.')
+            possible_clusters = self.get_clusters(clusters, cluster_fails, platform, retry_at)
+            for cluster in possible_clusters or []:
```

Comment:
Instead of `or []` get clusters should return `[]` if no clusters found in `get_clusters`